modin/core/io/text/excel_dispatcher.py

```diff
@@ -132,22 +125,38 @@ def _read(cls, io, **kwargs):
             while end_of_row_tag not in sheet_block:
                 sheet_block += f.read(EXCEL_READ_BLOCK_SIZE)
             idx_of_header_end = sheet_block.index(end_of_row_tag) + len(end_of_row_tag)
-            sheet_header = sheet_block[:idx_of_header_end]
-            # Reset the file pointer to begin at the end of the header information.
-            f.seek(idx_of_header_end)
+            sheet_header_with_first_row = sheet_block[:idx_of_header_end]
+
+            if not kwargs["header"] is None:
```

Comment:
```suggestion
            if kwargs["header"] is not None:
```
I personally like this form (`foo is not bar` vs `not foo is bar`) more, as I feel it's more natural to read